src/ucp/core/ucp_ep.c

```diff
@@ -3812,6 +3812,70 @@ ucs_status_t ucp_ep_realloc_lanes(ucp_ep_h ep, unsigned new_num_lanes)
     return UCS_OK;
 }
 
+static void
+ucp_ep_select_short_init(ucp_ep_h ep, unsigned feature_flag,
+                         ucp_operation_id_t op_id, unsigned proto_flags,
+                         ucp_lane_index_t exp_lane,
+                         ucp_memtype_thresh_t *max_eager_short)
+{
+    ucp_ep_config_t *ep_config = ucp_ep_config(ep);
+    ucp_proto_select_short_t proto_short;
+
+    if (ep->worker->context->config.features & feature_flag) {
+        ucp_proto_select_short_init(ep->worker, &ep_config->proto_select,
+                                    ep->cfg_index, UCP_WORKER_CFG_INDEX_NULL,
+                                    op_id, proto_flags, &proto_short);
+
+        /* Short protocol should be either disabled, or use expected lane */
+        ucs_assertv((proto_short.max_length_host_mem < 0) ||
+                    (proto_short.lane == exp_lane),
+                    "max_length_host_mem %ld, lane %d",
+                    proto_short.max_length_host_mem, proto_short.lane);
+    } else {
+        ucp_proto_select_short_disable(&proto_short);
+    }
+
+    max_eager_short->memtype_off = proto_short.max_length_unknown_mem;
+    max_eager_short->memtype_on  = proto_short.max_length_host_mem;
+}
+
+static void ucp_ep_select_short_init_all(ucp_ep_h ep)
```

Comment:
1. rename it to ucp_ep_config_proto_init
2. remove ep parameter. pass only worker and cfg_index
3. check proto_enable inside this function